src/Compilers/CSharp/Portable/Symbols/VarianceSafety.cs

```diff
@@ -115,7 +115,7 @@ private static void CheckPropertyVarianceSafety(PropertySymbol property, Diagnos
                 IsVarianceUnsafe(
                     property.Type,
                     requireOutputSafety: hasGetter,
-                    requireInputSafety: hasSetter,
+                    requireInputSafety: hasSetter || !(property.GetMethod?.RefKind == RefKind.None),
```

Comment:
would `!=` be true if there is no getter?

---
In reply to: [101596322](https://github.com/dotnet/roslyn/pull/17179#discussion_r101596322) [](ancestors = 101596322)